src/lib/util/zippath.cpp

```diff
@@ -111,9 +111,9 @@ int is_path_separator(char c)
  * @param [in,out]  endpos      If non-null, the endpos.
  */
 
-static void parse_parent_path(const char *path, int *beginpos, int *endpos)
+static void parse_parent_path(const std::string &path, int *beginpos, int *endpos)
 {
-	int length = strlen(path);
+	auto length = path.length();
```

Comment:
Now that you've got auto here you've made a bunch of singed/unsigned comparisons. Really, the beginpos/endpos parameters should be changed to `std::string::size_type` and `std::string::npos` can be used as a sentinel rather than a negative value, or they should be changed to iterators and an end iterator can be used as a sentinel.